lib/proxy-agent.js

```diff
@@ -136,14 +134,12 @@ class ProxyAgent extends DispatcherBase {
     )
   }
 
-  async [kClose] () {
-    await this[kAgent].close()
-    await this[kClient].close()
+  [kClose] () {
+    return Promise.all([this[kAgent].close, this[kClient].close])
   }
 
-  async [kDestroy] () {
-    await this[kAgent].destroy()
-    await this[kClient].destroy()
+  [kDestroy] () {
+    return Promise.all([this[kAgent].destroy, this[kClient].destroy])
```

Comment:
```suggestion
    return Promise.all([this[kAgent].destroy(), this[kClient].destroy()])
```
are they functions, they aren't?